pypeit/images/rawimage.py

```diff
@@ -19,399 +19,701 @@
+        :attr:`par`), in the order they will be applied are:
+
+            #. :func:`subtract_pattern`: Analyze and subtract the pattern noise
+               from the image.
+
+            #. :func:`build_rn2img`: Construct the detector variance image,
+               which includes readnoise and digitization error.  If any of the
+               amplifiers on the detector do not have a measured readnoise, the
+               readnoise is estimated using :func:`estimate_readnoise`.
+            
+            #. :func:`subtract_overscan`: Analyze the overscan region and
+               subtract from the image
+
+            #. :func:`trim`: Trim the image down to the data (i.e. remove the
+               overscan)
 
-        Note:  The processing step order is currently 'frozen' as is.
-          We may choose to allow optional ordering
+            #. :func:`orient`: Orient the image in the PypeIt orientation ---
+               spectral coordinates ordered along the first axis and spatial
+               coordinates ordered along the second, ``(spec, spat)`` --- with
+               blue to red going from small pixel numbers to large.
 
-        Here are the allowed steps, in the order they will be applied:
-            subtract_overscan -- Analyze the overscan region and subtract from the image
-            trim -- Trim the image down to the data (i.e. remove the overscan)
-            orient -- Orient the image in the PypeIt orientation (spec, spat) with blue to red going down to up
-            subtract_bias -- Subtract a bias image
-            apply_gain -- Convert to counts, amp by amp
-            flatten -- Divide by the pixel flat and (if provided) the illumination flat
-            extras -- Generate the RN2 and IVAR images
-            crmask -- Generate a CR mask
+            #. :func:`subtract_bias`: Subtract a bias image.  The shape and
+               orientation of the bias image must match the *processed* image.
+               I.e., if you trim and orient this image, you must also have
+               trimmed and oriented the bias frames.  The units of the bias
+               image *must* be ADU.
+
+            #. :func:`apply_gain`: Convert from ADU to electrons, amp by amp.
+
+            #. :func:`subtract_dark`: Subtract a dark image.  The shape and
+               orientation of the dark image must match the *processed* image.
+               I.e., if you trim and orient this image, you must also have
+               trimmed and oriented the dark frames.  The units of the dark
+               frame *must* be electrons (counts).
+
+            #. :func:`add_shot_noise`: Include shot-noise from the image
+               variance.
+
+            #. :func:`impose_noise_floor`: Add uncertainty to the variance image
+               such that the S/N per pixel is never reaches higher than
+               ``1/noise_floor``.
+
+            #. :func:`spatial_flexure_shift`: Measure any spatial shift due to flexure.
```

Comment:
I don't think this measures a flexure shift, I believe it just applies it.

Reply:
No, it actually does call `flexure.spat_flexure_shift`, which measures the shift.  Once the shift is measured, the flat-fielding function calls `flatimages.fit2illumflat`, which applies the shift.